modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/action/SharingFileEntryContentDashboardItemAction.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.content.dashboard.document.library.internal.item.action;
+
+import com.liferay.content.dashboard.item.action.ContentDashboardItemAction;
+import com.liferay.petra.portlet.url.builder.PortletURLBuilder;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.portlet.LiferayPortletResponse;
+import com.liferay.portal.kernel.portlet.LiferayWindowState;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.util.JavaConstants;
+import com.liferay.portal.kernel.util.Portal;
+
+import java.util.Locale;
+
+import javax.portlet.PortletResponse;
+import javax.portlet.PortletURL;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Cristina González
+ */
+public class SharingFileEntryContentDashboardItemAction
+	implements ContentDashboardItemAction {
+
+	public SharingFileEntryContentDashboardItemAction(
+		FileEntry fileEntry, HttpServletRequest httpServletRequest,
+		Language language, Portal portal) {
+
+		_fileEntry = fileEntry;
+		_httpServletRequest = httpServletRequest;
+		_language = language;
+		_portal = portal;
+	}
+
+	@Override
+	public String getIcon() {
+		return "share";
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "share");
+	}
+
+	@Override
+	public String getName() {
+		return "share";
+	}
+
+	@Override
+	public Type getType() {
+		return Type.SHARING;
+	}
+
+	@Override
+	public String getURL() {
+		try {
+			PortletResponse portletResponse =
+				(PortletResponse)_httpServletRequest.getAttribute(
+					JavaConstants.JAVAX_PORTLET_RESPONSE);
+
+			LiferayPortletResponse liferayPortletResponse =
+				_portal.getLiferayPortletResponse(portletResponse);
+
+			PortletURL portletURL = PortletURLBuilder.createRenderURL(
+				liferayPortletResponse,
+				"com_liferay_content_dashboard_web_portlet_" +
+				"ContentDashboardAdminPortlet"
+			).setMVCPath(
+				"/sharing_button.jsp"
+			).setParameter(
+				"className", FileEntry.class.getName()
```

Comment:
Hey @Stephy6 !!!

Have you tried to use DLFileEntry.class.getName() here?

Thanks!!

Reply:
Awesome!!, do you have any problem in the Logs?

Thanks :)